hedera-node/hedera-app/src/main/java/com/hedera/node/app/HederaApp.java

```diff
@@ -0,0 +1,118 @@
+import com.hedera.node.app.service.mono.context.properties.*;
+import com.hedera.node.app.service.mono.contracts.ContractsModule;
+import com.hedera.node.app.service.mono.fees.FeesModule;
+import com.hedera.node.app.service.mono.files.FilesModule;
+import com.hedera.node.app.service.mono.grpc.GrpcModule;
+import com.hedera.node.app.service.mono.keys.KeysModule;
+import com.hedera.node.app.service.mono.ledger.LedgerModule;
+import com.hedera.node.app.service.mono.queries.QueriesModule;
+import com.hedera.node.app.service.mono.records.RecordsModule;
+import com.hedera.node.app.service.mono.sigs.SigsModule;
+import com.hedera.node.app.service.mono.state.StateModule;
+import com.hedera.node.app.service.mono.state.expiry.ExpiryModule;
+import com.hedera.node.app.service.mono.state.tasks.TaskModule;
+import com.hedera.node.app.service.mono.stats.StatsModule;
+import com.hedera.node.app.service.mono.store.StoresModule;
+import com.hedera.node.app.service.mono.throttling.ThrottlingModule;
+import com.hedera.node.app.service.mono.txns.TransactionsModule;
+import com.hedera.node.app.service.mono.txns.submission.SubmissionModule;
+import com.hedera.node.app.services.ServiceModule;
+import com.hedera.node.app.workflows.query.QueryModule;
+import com.swirlds.common.crypto.Cryptography;
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.system.Platform;
+import dagger.BindsInstance;
+import dagger.Component;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+/** The infrastructure used to implement the platform contract for a Hedera Services node.
+ * This is needed for adding dagger subcomponents.
+ * Currently, it extends {@link com.hedera.node.app.service.mono.ServicesApp}. But,
+ * in the future this class will be cleaned up to not have multiple module dependencies */
+@Singleton
+@Component(
+        modules = {
+            TaskModule.class,
+            FeesModule.class,
+            KeysModule.class,
+            SigsModule.class,
+            GrpcModule.class,
+            ConfigModule.class,
+            StatsModule.class,
+            StateModule.class,
+            FilesModule.class,
+            LedgerModule.class,
+            StoresModule.class,
+            ContextModule.class,
+            RecordsModule.class,
+            QueriesModule.class,
+            ContractsModule.class,
+            PropertiesModule.class,
+            ThrottlingModule.class,
+            SubmissionModule.class,
+            TransactionsModule.class,
+            ExpiryModule.class,
+            ServiceModule.class,
+            QueryModule.class
+        })
+public interface HederaApp extends ServicesApp {
+    /* Needed by ServicesState */
+
+    Provider<QueryComponent.Factory> queryComponentFactory();
+
+    @Component.Builder
+    interface Builder {
+        @BindsInstance
+        Builder crypto(Cryptography engine);
```

Comment:
Please use `com.swirlds.common.contextPlatformContext`. This will provide access to crypto, config + metrics

Reply:
I see. Did not know that. Will chnage to that

Reply:
https://github.com/hashgraph/hedera-services/issues/5030